syntax = "proto3";

package brain.shared.v1;

import "brain/shared/v1/envelope.proto";

service CoreHealthService {
  rpc Health(CoreHealthRequest) returns (CoreHealthResponse);
}

message CoreComponentHealthStatus {
  bool ready = 1;
  string detail = 2;
}

message CoreHealthStatus {
  bool ready = 1;
  map<string, CoreComponentHealthStatus> services = 2;
  map<string, CoreComponentHealthStatus> resources = 3;
}

message CoreHealthPayload {}

message CoreHealthRequest {
  EnvelopeMeta metadata = 1;
  CoreHealthPayload payload = 2;
}

message CoreHealthResponse {
  EnvelopeMeta metadata = 1;
  CoreHealthStatus payload = 2;
  repeated ErrorDetail errors = 3;
}
